canonical-web-and-design/maas-ui
# fix(legacy): fix removing domains with records

## Done

- Delete dns resources after they've been removed from a domain so that the domain can be deleted.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the DNS page.
- Add a domain.
- Go to the domain details page.
- Add an A or AAAAAA record.
- Add a record that is not an A or AAAAAAAAAAA.
- Remove both records.
- Delete the domain.
- It should delete successfully.

## Fixes

Fixes: #1892.
